<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ul.lst-kix_ikaq0minelf9-1{list-style-type:none}ul.lst-kix_ikaq0minelf9-2{list-style-type:none}.lst-kix_ikaq0minelf9-8>li:before{content:"\0025a0   "}ul.lst-kix_ikaq0minelf9-0{list-style-type:none}.lst-kix_ikaq0minelf9-2>li:before{content:"\0025a0   "}.lst-kix_ikaq0minelf9-3>li:before{content:"\0025cf   "}.lst-kix_ikaq0minelf9-1>li:before{content:"\0025cb   "}.lst-kix_ikaq0minelf9-5>li:before{content:"\0025a0   "}.lst-kix_ikaq0minelf9-0>li:before{content:"  "}.lst-kix_ikaq0minelf9-4>li:before{content:"\0025cb   "}ul.lst-kix_ikaq0minelf9-5{list-style-type:none}.lst-kix_ikaq0minelf9-7>li:before{content:"\0025cb   "}ul.lst-kix_ikaq0minelf9-6{list-style-type:none}ul.lst-kix_ikaq0minelf9-3{list-style-type:none}ul.lst-kix_ikaq0minelf9-4{list-style-type:none}.lst-kix_ikaq0minelf9-6>li:before{content:"\0025cf   "}ul.lst-kix_ikaq0minelf9-7{list-style-type:none}ul.lst-kix_ikaq0minelf9-8{list-style-type:none}ol{margin:0;padding:0}table td,table th{padding:0}.c6{padding-top:20pt;padding-bottom:6pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c7{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:20pt;font-family:"Arial";font-style:normal}.c4{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c1{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c5{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c8{max-width:468pt;padding:72pt 72pt 72pt 72pt}.c0{color:inherit;text-decoration:inherit}.c2{background-color:#ffffff}.c3{height:11pt}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c2 c8 doc-content"><h1 class="c6" id="h.986c7pzdvgr0"><span class="c7">Description</span></h1><p class="c1"><span>Multiplex assays of variant effect (MAVEs) can test thousands of sequence variants in a single experiment. To facilitate the application of MAVE experimental results, researchers with the Alliance of Variant Effects Alliance (AVE) have developed MaveDB ( </span><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://www.mavedb.org&amp;sa=D&amp;source=editors&amp;ust=1712111513922916&amp;usg=AOvVaw39VLWBQaTjR7OjKo9A_Yf1">https://www.mavedb.org</a></span><span class="c4">&nbsp;), a public repository for large-scale measurements of sequence variant impact. &nbsp;While these measurements have strong potential to accelerate sequence variant interpretations, the initial lack of reporting standards has hindered their use. &nbsp;In particular, the practice of reporting MAVE results relative to assayed sequences rather than accessioned genomic, transcript, or protein coordinates has complicated their use. &nbsp;In response, the Data Coordination and Dissemination Workstream of the AVE has developed new methods to map MAVE data to human reference sequences. &nbsp;These tracks present hundreds of MAVE experiments mapped to the human reference genome.</span></p><h1 class="c6" id="h.kpklhyqaux3"><span class="c7">Genomes</span></h1><p class="c1"><span class="c4">We provide MAVE data mappings on the GRCh38 genome.</span></p><h1 class="c6" id="h.mjc1249x4n1u"><span class="c7">Display Conventions and Configuration</span></h1><p class="c1"><span class="c4">Each MAVE score set is represented as a heat map, with rows representing alternative alleles and columns representing genomic positions. &nbsp;Each cell in the heat map is colored according to the MAVE score, with the highest scores shown in yellow and the lowest scores shown in purple. &nbsp;Further data on each experiment is available through MaveDB. </span></p><h1 class="c6" id="h.7di0evggxb85"><span class="c7">Methods</span></h1><p class="c1"><span>Variant mapping from human score sets was accomplished in three sequential steps. &nbsp;First, each MaveDB target sequence was aligned to the GRCh38 human genome assembly using BLAT. &nbsp;Second, we identified compatible transcript or protein reference sequences from RefSeq based on overlapping coordinates, leveraging the &nbsp;biocommons SeqRepo python package, the Universal Transcript Archive (UTA) database and GenomicMedLab Common Operations on Lots of Sequences (cool-seq-tool) python package. For MAVE score sets which could be mapped directly to a contiguous chromosomal region, such as those which interrogate regulatory or other noncoding elements, this step was skipped. &nbsp;Lastly, we extracted the genomic coordinates to assemble genome browser tracks, using the UCSC Genome Browser&rsquo;s BigHeat utility as adapted within an </span><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://github.com/BRCAChallenge/mavedb&amp;sa=D&amp;source=editors&amp;ust=1712111513923832&amp;usg=AOvVaw1RsLzicQ0cRQ7BKAj1wKmh">open source python software package</a></span><span class="c4">.</span></p><h1 class="c6" id="h.lh4mydunil7l"><span class="c7">Data Availability</span></h1><p class="c1"><span>All data are freely available for download via the MaveDB FastAPI-driven API, specified at </span><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://api.mavedb.org/docs&amp;sa=D&amp;source=editors&amp;ust=1712111513924133&amp;usg=AOvVaw3MMpYQRwdTTWksatfVE01b">https://api.mavedb.org/docs</a></span><span class="c4">&nbsp; </span></p><h1 class="c6" id="h.xukp72k0k5sd"><span class="c7">Credits</span></h1><p class="c1"><span class="c2">MaveDB is hosted by the </span><span class="c5 c2"><a class="c0" href="https://www.google.com/url?q=https://faculty.washington.edu/dfowler/index.html&amp;sa=D&amp;source=editors&amp;ust=1712111513924563&amp;usg=AOvVaw1KpYxKXVAZYPtWFwwK0bEu">Fowler Lab</a></span><span class="c2">&nbsp;in the </span><span class="c5 c2"><a class="c0" href="https://www.google.com/url?q=https://www.gs.washington.edu/&amp;sa=D&amp;source=editors&amp;ust=1712111513924733&amp;usg=AOvVaw2DtB4v5YFC-rrrgOKdqi6s">Department of Genome Sciences</a></span><span class="c2">&nbsp;at the University of Washington. It is supported and developed by the </span><span class="c5 c2"><a class="c0" href="https://www.google.com/url?q=https://www.washington.edu/&amp;sa=D&amp;source=editors&amp;ust=1712111513924911&amp;usg=AOvVaw18pf9SLa-M6K1L0giNdLMV">University of Washington</a></span><span class="c2">, the </span><span class="c5 c2"><a class="c0" href="https://www.google.com/url?q=https://www.wehi.edu.au/&amp;sa=D&amp;source=editors&amp;ust=1712111513925111&amp;usg=AOvVaw1lgMPxN22GWlLmyErFTscU">Walter and Eliza Hall Institute of Medical Research</a></span><span class="c2">, and the </span><span class="c5 c2"><a class="c0" href="https://www.google.com/url?q=https://brotmanbaty.org/&amp;sa=D&amp;source=editors&amp;ust=1712111513925294&amp;usg=AOvVaw2LoHJk342S8qO_bh92jc5r">Brotman Baty Institute.</a></span></p><h1 class="c6" id="h.vs6575nqzuat"><span class="c7">References</span></h1><p class="c1"><span>Esposito D, Weile J, Shendure J, Starita LM, Papenfuss AT, Roth FP, Fowler DM, Rubin AF. &nbsp;</span><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://pubmed.ncbi.nlm.nih.gov/31679514/&amp;sa=D&amp;source=editors&amp;ust=1712111513925742&amp;usg=AOvVaw2XlP06K61IgyHxDAxvfgpi">MaveDB: an open-source platform to distribute and interpret data from multiplexed assays of variant effect,</a></span><span>&nbsp;Genome Biol. 2019 Nov 4;20(1):223. doi: 10.1186/s13059-019-1845-6. PMID: </span><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://pubmed.ncbi.nlm.nih.gov/31679514/&amp;sa=D&amp;source=editors&amp;ust=1712111513925919&amp;usg=AOvVaw2FQg-FoFdPScVXg8pBztwM">31679514</a></span><span>&nbsp;PMCID: </span><span class="c5"><a class="c0" href="https://www.google.com/url?q=http://www.ncbi.nlm.nih.gov/pmc/articles/pmc6827219/&amp;sa=D&amp;source=editors&amp;ust=1712111513926157&amp;usg=AOvVaw2UqlW15iQxSWAZP05sRyJE">PMC6827219</a></span><span>&nbsp;DOI: </span><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://doi.org/10.1186/s13059-019-1845-6&amp;sa=D&amp;source=editors&amp;ust=1712111513926354&amp;usg=AOvVaw06KUo_kxRY4cFpTmgaui0a">10.1186/s13059-019-1845-6</a></span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1"><span class="c4">Arbesfeld JA, Da EY, Kuzma K, Paul A, Farris T, Riehle K, Saraiva Agostinho ND, Safer JF, Milosavljevic A, Foreman J, Firth HV, Hunt SE, Iqbal S, Cline MS, Rubin AF, Wagner AH. </span></p><p class="c1"><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://www.biorxiv.org/content/10.1101/2023.06.20.545702v1&amp;sa=D&amp;source=editors&amp;ust=1712111513926679&amp;usg=AOvVaw3CINvXKd28tiwczlWt9ss_">Mapping MAVE data for use in human genomics applications</a></span><span>. bioRxiv </span><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://www.biorxiv.org/content/10.1101/2023.06.20.545702v1&amp;sa=D&amp;source=editors&amp;ust=1712111513926835&amp;usg=AOvVaw1nlI-BM6fdyI87dLsXeM4z">2023.06.20.545702</a></span><span>; doi: </span><span class="c5"><a class="c0" href="https://www.google.com/url?q=https://doi.org/10.1101/2023.06.20.545702&amp;sa=D&amp;source=editors&amp;ust=1712111513926973&amp;usg=AOvVaw2GDIMb9hkoi0_E7zh3APgV">https://doi.org/10.1101/2023.06.20.545702</a></span><span class="c4">.</span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1 c3"><span class="c4"></span></p><p class="c1 c3"><span class="c4"></span></p></body></html>